tests/test_nhg.py

```diff
@@ -10,30 +10,115 @@
 
 
 class TestNextHopGroup(object):
-    def test_route_nhg(self, dvs, dvs_route, testlog):
+    ASIC_NHS_STR = "ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP"
+    ASIC_NHG_STR = "ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP_GROUP"
+    ASIC_NHGM_STR = ASIC_NHG_STR + "_MEMBER"
+    ASIC_RT_STR = "ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY"
+    ASIC_INSEG_STR = "ASIC_STATE:SAI_OBJECT_TYPE_INSEG_ENTRY"
+
+    def get_route_id(self, prefix, dvs):
+        for k in dvs.get_asic_db().get_keys(self.ASIC_RT_STR):
+            if json.loads(k)['dest'] == prefix:
+                return k
+
+        return None
+
+    def get_inseg_id(self, label, dvs):
+        for k in dvs.get_asic_db().get_keys(self.ASIC_INSEG_STR):
+            print(json.loads(k))
+            if json.loads(k)['label'] == label:
+                return k
+
+        return None
+
+    def get_nhg_id(self, nhg_index, dvs):
+        # Add a route with the given index, then retrieve the next hop group ID
+        # from that route
+        asic_db = dvs.get_asic_db()
+        asic_rts_count = len(asic_db.get_keys(self.ASIC_RT_STR))
+
+        fvs = swsscommon.FieldValuePairs([('nexthop_group', nhg_index)])
+        prefix = '255.255.255.255/24'
```

Comment:
why is there a need to add a 'prefix'?

Reply:
In order to get the SAI ID of a NHG, we're adding a route pointing to that NHG given the NHG's key, then we get the ASIC route's NEXT_HOP_ID. Otherwise we have no easier way of finding the SAI ID of a NHG given only its APPL_DB key.

Reply:
Just for my uderstanding, won't nhgOrch calling "create_next_hop_group()" result in creating the SAI ID of the NHG. I don't know the details but it looks sufficient to add the nexthop group to get the SAI ID.

Reply:
I see what you mean. This is fine.